lib/string_extras.rb

```diff
@@ -207,11 +210,15 @@ def substitute!(pattern, replacement)
   # When strings are mistakenly encoded as single-byte character sets, instead
   # of UTF-8, there are some distinctive character combinations that we can spot
   # and fix
+  # Useful table here http://www.i18nqa.com/debug/utf8-debug.html
   BAD_ENCODING = {
     'â‚¬' => '€', 'â€š' => '‚', 'Æ’' => 'ƒ', 'â€ž' => '„', 'â€¦' => '…',
     'â€ ' => '†', 'â€¡' => '‡', 'Ë†' => 'ˆ', 'â€°' => '‰', 'Å ' => 'Š',
     'â€¹' => '‹', 'Å’' => 'Œ', 'Å½' => 'Ž', 'â€˜' => '‘', 'â€™' => '’',
-    'â€œ' => '“', 'â€�' => '”', 'â€¢' => '•', 'â€“' => '–', 'â€”' => '—',
+    'â€œ' => '“',
+    'â€' => '”', # Note the invisible Ux009D in the key
+    'â€²' => '′', # Manually added. Some seem to use this instead of Ux2019
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.